src/driver/interface/mailstorage.c

```diff
@@ -320,7 +320,7 @@ void mailstorage_disconnect(struct mailstorage * storage)
 {
   clistiter * cur;
 
-  while ((cur = clist_begin(storage->sto_shared_folders)) != NULL) {
+  for (cur = clist_begin(storage->sto_shared_folders); cur != NULL; cur = clist_next(cur)) {
```

Comment:
mailfolder_disconnect() is probably touching the data structure storage->sto_shared_folders... the new version might lead to some inconsistencies / crashes.

Reply:
Well, the alert is the loop break don't depend interations.
I think the loop can run forever as it is written.